cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/sql/PostgreSqlStructuredTableAdmin.java

```diff
@@ -151,8 +152,7 @@ private void createTable(StructuredTableSpecification spec) throws IOException {
         }
 
         // Create indexes
-        Set<String> nonExistIndexes = getNonExistIndexes(connection, new StructuredTableSchema(spec));
-        for (String indexStatement : getCreateIndexStatements(spec.getTableId(), nonExistIndexes)) {
+        for (String indexStatement : getCreateIndexStatements(spec.getTableId(), new HashSet<>(spec.getIndexes()))) {
```

Comment:
Why this change? I mean, it would work, but you would issue much more "Create index" statements.

Reply:
This is creating table senario, the table does not exist yet

Reply:
Okay this can be edge case that it's being created at the same time by two instances... will revert back using DB query result

Reply:
ref: https://github.com/cdapio/cdap/pull/14670/files#r1001251902

Reply:
I mean, it's trading one query (select) for potentially multiple (create index). It's probably fine as it's corner case and it's safe anyway